scripts/install.sh

```diff
@@ -1,490 +1,56 @@
-        _cputype="$RUSTUP_CPUTYPE"
-      else {
-        # 32-bit executable for amd64 = x32
-        if is_host_amd64_elf; then
-          err "x86_64 linux with x86 userland unsupported"
-        else
-          _cputype=i686
-        fi
-      }; fi
-      ;;
-    mips64)
-      _cputype=$(get_endianness mips '' el)
-      ;;
-    powerpc64)
-      _cputype=powerpc
-      ;;
-    aarch64)
-      _cputype=armv7
-      if [ "$_ostype" = "linux-android" ]; then
-        _ostype=linux-androideabi
-      else
-        _ostype="${_ostype}eabihf"
-      fi
-      ;;
-    riscv64gc)
-      err "riscv64 with 32-bit userland unsupported"
-      ;;
-    esac
-  fi
-
-  # Detect armv7 but without the CPU features Rust needs in that build,
-  # and fall back to arm.
-  # See https://github.com/rust-lang/rustup.rs/issues/587.
-  if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
-    if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
-      # At least one processor does not have NEON.
-      _cputype=arm
-    fi
-  fi
-
-  _arch="${_cputype}-${_ostype}"
-
-  RETVAL="$_arch"
-}
-
-resolve_version() {
-  local _requested_version=$1
-  local _requested_ref=$2
-
-  local _response
-
-  say "retrieving $_requested_version version from ${REPO}..."
-  _response=$(ensure curl -Ls -H 'Accept: application/json' "${REPO}/releases/${_requested_ref}")
-  if [ "{\"error\":\"Not Found\"}" = "$_response" ]; then
-    err "version $_requested_version not found"
-  fi
-
-  RETVAL=$(echo "$_response" | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
-}
-
-create_install_dir() {
-  local _requested_version=$1
-
-  local _installdir="${INSTALL_ROOT}/${_requested_version}"
-
-  if [ -d "$_installdir" ]; then
-    ensure rm -rf "$_installdir"
-    say "removed existing snforge and sncast installation at ${_installdir}"
-  fi
-
-  ensure mkdir -p "$_installdir"
-
-  RETVAL=$_installdir
-}
-
-download() {
-  local _resolved_version=$1
-  local _arch=$2
-  local _installdir=$3
-  local _tempdir=$4
-
-  local _tarball="starknet-foundry-${_resolved_version}-${_arch}.tar.gz"
-  local _url="${REPO}/releases/download/${_resolved_version}/${_tarball}"
-  local _dl="$_tempdir/starknet-foundry.tar.gz"
-
-  say "downloading ${_tarball}..."
-
-  ensure curl -fLS -o "$_dl" "$_url"
-  ensure tar -xz -C "$_installdir" --strip-components=1 -f "$_dl"
-}
-
-create_symlinks() {
-  for binary in "snforge" "sncast"; do
-    local _binary="${_installdir}/bin/${binary}"
-    local _symlink="${LOCAL_BIN}/${binary}"
-
-    ensure mkdir -p "$LOCAL_BIN"
-    ensure ln -fs "$_binary" "$_symlink"
-    ensure chmod u+x "$_symlink"
-    say "created symlink ${_symlink} -> ${_binary}"
-  done
-}
+# Only add snfoundryup if it isn't already in PATH.
+if [ ":$PATH:" != *":${LOCAL_BIN}:"* ]; then
+    # Add the snfoundryup directory to the path and ensure the old PATH variables remain.
+    echo >> "$PROFILE" && echo "export PATH=\"\$PATH:$LOCAL_BIN\"" >> "$PROFILE"
+fi
```

Comment:
I think this might get executed each time this script is being called, as (if I remember correctly) test command (`[ ... ]`) cannot match globs. 

edit: yes, it looks that way:
```
wojciech@mbp ~/Tests $ ➜ echo $PATH | grep .local
(...):/Users/wojciech/.local/bin

wojciech@mbp ~/Tests $ ➜ echo $0
-zsh

wojciech@mbp ~/Tests $ ➜ cat ~/.zshenv
. "$HOME/.cargo/env"

export PATH="$PATH:/Users/wojciech/.local/bin"

wojciech@mbp ~/Tests $ ➜ ./install.sh
Installing snfoundryup...
################################################################################################################################################################################################################################################### 100.0%

Detected your preferred shell is zsh and added snfoundryup to PATH. Run 'source /Users/wojciech/.zshenv' or start a new terminal session to use snfoundryup.
Then, simply run 'snfoundryup' to install Starknet-Foundry.

wojciech@mbp ~/Tests $ ➜ cat ~/.zshenv
. "$HOME/.cargo/env"

export PATH="$PATH:/Users/wojciech/.local/bin"

export PATH="$PATH:/Users/wojciech/.local/bin"
```

Can you fix please? 🙏

Reply:
good catch, should be fixed now.